pkg/commands/domainv1/tools/suggest.go

```diff
@@ -0,0 +1,130 @@
+package tools
+
+import (
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/fastly/cli/pkg/argparser"
+	fsterr "github.com/fastly/cli/pkg/errors"
+	"github.com/fastly/cli/pkg/global"
+	"github.com/fastly/cli/pkg/text"
+	"github.com/fastly/go-fastly/v10/fastly"
+	"github.com/fastly/go-fastly/v10/fastly/domains/v1/tools/suggest"
+)
+
+// GetDomainSuggestionsCommand calls the Fastly API to perform a real-time query of the search term(s) against the
+// know zones database.
+type GetDomainSuggestionsCommand struct {
+	argparser.Base
+	argparser.JSONOutput
+	// Required.
+	query string
+	// Optional.
+	defaults argparser.OptionalString
+	keywords argparser.OptionalString
+	location argparser.OptionalString
+	vendor   argparser.OptionalString
+}
+
+// NewDomainSuggestionsCommand returns a usable DomainSuggestionCommand registered under the parent.
+func NewDomainSuggestionsCommand(parent argparser.Registerer, g *global.Data) *GetDomainSuggestionsCommand {
+	cmd := GetDomainSuggestionsCommand{
+		Base: argparser.Base{
+			Globals: g,
+		},
+	}
+
+	cmd.CmdClause = parent.Command("suggest", "Performs real-time queries against the known zones database")
+	// Required.
+	cmd.CmdClause.Flag("query", "The term(s) to search against.").Required().StringVar(&cmd.query)
+	// Optional.
+	cmd.CmdClause.Flag("defaults", "Comma-separated list of default zones to include in the search results response").Action(cmd.defaults.Set).StringVar(&cmd.defaults.Value)
+	cmd.CmdClause.Flag("keywords", "Comma-separated list of keywords for seeding the results").Action(cmd.keywords.Set).StringVar(&cmd.keywords.Value)
+	cmd.CmdClause.Flag("location", "Overrides the IP location detection for country-code zones, with a two-character country code").Action(cmd.location.Set).StringVar(&cmd.location.Value)
+	cmd.CmdClause.Flag("vendor", "The domain name of a specific registrar or vendor ").Action(cmd.vendor.Set).StringVar(&cmd.vendor.Value)
+	cmd.RegisterFlagBool(cmd.JSONFlag())
```

Comment:
Re-ordered flags so they're displayed in the help output lexicographically (basically just moved the JSON flag):

```suggestion
	cmd.CmdClause.Flag("defaults", "Comma-separated list of default zones to include in the search results response").Action(cmd.defaults.Set).StringVar(&cmd.defaults.Value)
	cmd.RegisterFlagBool(cmd.JSONFlag())
	cmd.CmdClause.Flag("keywords", "Comma-separated list of keywords for seeding the results").Action(cmd.keywords.Set).StringVar(&cmd.keywords.Value)
	cmd.CmdClause.Flag("location", "Overrides the IP location detection for country-code zones, with a two-character country code").Action(cmd.location.Set).StringVar(&cmd.location.Value)
	cmd.CmdClause.Flag("vendor", "The domain name of a specific registrar or vendor ").Action(cmd.vendor.Set).StringVar(&cmd.vendor.Value)
```